pool/app/session/entity.ml

```diff
@@ -242,10 +251,28 @@ module Public = struct
     }
   [@@deriving eq, show]
 
+  let not_canceled (session : t) =
+    match session.canceled_at with
+    | None -> Ok ()
+    | Some canceled_at -> is_canceled_error canceled_at
+  ;;
+
   let is_fully_booked (m : t) =
     m.assignment_count >= m.max_participants + m.overbook
   ;;
 
+  let assignment_creatable session =
+    let open CCResult.Infix in
+    let* () =
+      is_fully_booked session
+      |> function
```

Comment:
Can we introduce a `bool_to_result_not` or similar? There are already several occurrences where we add a function to the `false` state and an error on the `true` value.

function signature:
`val bool_to_result_not : bool -> (unit, Pool_common.Message.error) result`